LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/MessageProcessor.cs

```diff
@@ -169,11 +169,15 @@ async Task<DownlinkPktFwdMessage> ProcessDataMessageAsync(LoRaTools.LoRaPhysical
                     var isConfirmedResubmit = false;
                     if (!isFrameCounterFromNewlyStartedDevice && payloadFcnt <= loRaDevice.FCntUp)
                     {
-                        // TODO: have a maximum retry (3)
-                        // Future: Keep track of how many times we acked the confirmed message (4+ times we skip)
                         // if it is confirmed most probably we did not ack in time before or device lost the ack packet so we should continue but not send the msg to iothub
                         if (requiresConfirmation && payloadFcnt == loRaDevice.FCntUp)
                         {
+                            if (!loRaDevice.ValidateConfirmResubmit(payloadFcnt))
+                            {
+                                Logger.Log(loRaDevice.DevEUI, $"resubmit from confirmed message exceeds threshold of {LoRaDevice.MaxConfirmationResubmitCount}, message ignored, msg: {payloadFcnt} server: {loRaDevice.FCntUp}", LogLevel.Information);
```

Comment:
Should be debug so we don't generate logs and log message that could be used as attack

Reply:
Done